python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,421 @@
+            }
+        )
+        self.delay = 1
+
+        self.ready_for_flats = False
+
+    async def warmup_lightsource(self, calib_type: str) -> None:
+        """Turn on the Chiller and the ATWhiteLight and wait for the
+        warmup to complete
+
+        Parameters
+        ----------
+        calib_type : `str`
+            Identifies if White Light or Monochromatic. For the ATCalsys,
+            the light sourceis the same. However, if it is White Light
+            configure the Monochromator now.
+
+        Raises
+        ------
+        RuntimeError:
+
+        """
+        self.calib_type = calib_type
+        try:
+            await self.start_chiller()
+            await self.wait_for_chiller_temp_within_tolerance()
+            await self.switch_lamp_on()
+            await self.wait_for_lamp_to_warm_up()
+            self.ready_for_flats = True
+        except asyncio.TimeoutError:
+            raise RuntimeError("ATWhitelight warmup issue.")
+
+    async def configure_projector(self, filter: str, wavelength: float) -> None:
+        """Configure the ATMonochromator according to the flat parameters
+
+        Parameters
+        ----------
+        filter : `str`
+            Determines which grating to use
+        wavelength : `float`
+            [unit: nm]
+            Output wavelength of the monochromator
+
+        Raises
+        ------
+        RuntimeError:
+
+        """
+        assert self.ready_for_flats
+        grating, entrance_slit, exit_slit = self._get_monochromator_setup(filter)
+
+        try:
+            await self.setup_monochromator(
+                wavelength, entrance_slit, exit_slit, grating
+            )
+        except asyncio.TimeoutError:
+            raise RuntimeError("Monochromator setup issue.")
+
+    async def setup_monochromator(
+        self, wavelength: float, entrance_slit: float, exit_slit: float, grating: int
+    ) -> None:
+        """Setup Monochromator.
+
+        Parameters
+        ----------
+        wavelength : `float`
+            Wavelength in nm.
+        entrance_slit : `float`
+            Size of entrance slit in mm.
+        exit_slit : `float`
+            Size of exist slit in mm.
+        grating : `int`
+            Grating to select.
+
+        """
+
+        await self.rem.atmonochromator.cmd_updateMonochromatorSetup.set_start(
+            wavelength=wavelength,
+            gratingType=grating,
+            fontExitSlitWidth=exit_slit,
+            fontEntranceSlitWidth=entrance_slit,
+            timeout=self.long_timeout,
+        )
+
+    async def electrometer_scan(self, duration: float) -> None:
+        """Perform an electrometer scan for the specified duration.
+
+        Parameters
+        ----------
+        duration : `float`
+            Total duration of scan.
+
+        Returns
+        -------
+        RunTimeError
+
+        """
+
+        self.rem.electrometer_1.evt_largeFileObjectAvailable.flush()
+
+        try:
```

Comment:
Again, I would remove the try/except here. However, this is a case where you might want to do a try/except around the `startScanDt` command and then, continue to wait for the LFOA event if the command timedout. This would certainly increase resiliency to missing acks. 

However, note that the correct exception to wait in these cases is not `asyncio.TimeoutError` but rather `salobj.AckTimeoutError`.

If you want to do something like this here I would recommend something like this:

```
try:
    await self.rem.electrometer_1.cmd_startScanDt.set_start(
                scanDuration=duration, timeout=duration + self.cmd_timeout
    )
except salobj.AckTimeoutError:
    self.log.exception("Timed out waiting for the command ack. Continuing.")

await self.rem.electrometer_1.evt_largeFileObjectAvailable.next(
                timeout=self.cmd_timeout, flush=False
            )
```

Also, please adjust the timeouts to use the ones provided by the base class as I mentioned above. 

Another thing to consider here is returning the LFOA event. This might be useful.

Reply:
Ok, done for electrometer and fiber spectrograph